### PR TITLE
Added Copy and Move constructors to Multiply Layers

### PR DESCRIPTION
**Pull request by Abilityguy**
Reference Issue #2625.
Changes to file ```rnn_impl.hpp``` remove the warning ```x will be initialised after y``` during ```make```.

### CODE REVIEW

**Comment by Abilityguy** [comment]
Hi @zoq , could you give this a review?
I believe it is ready for merge.

File: src/mlpack/methods/ann/layer/multiply_constant.hpp [hunk 1]
```diff
@@ -39,6 +39,18 @@ class MultiplyConstant
    */
   MultiplyConstant(const double scalar = 1.0);
 
+  //! Copy Constructor
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/multiply_constant.hpp:42]:
```suggestion
  //! Copy Constructor.
```

File: src/mlpack/methods/ann/layer/multiply_constant.hpp [hunk 2]
```diff
@@ -39,6 +39,18 @@ class MultiplyConstant
...
+  MultiplyConstant(const MultiplyConstant& layer);
+
+  //! Move Constructor
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/multiply_constant.hpp:45]:
```suggestion
  //! Move Constructor.
```

File: src/mlpack/methods/ann/layer/multiply_constant.hpp [hunk 3]
```diff
@@ -39,6 +39,18 @@ class MultiplyConstant
...
+  MultiplyConstant(MultiplyConstant&& layer);
+
+  //! Copy assignment operator
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/multiply_constant.hpp:48]:
```suggestion
  //! Copy assignment operator.
```

File: src/mlpack/methods/ann/layer/multiply_constant.hpp [hunk 4]
```diff
@@ -39,6 +39,18 @@ class MultiplyConstant
...
+  MultiplyConstant& operator=(const MultiplyConstant& layer);
+
+  //! Move assignment operator
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/multiply_constant.hpp:51]:
```suggestion
  //! Move assignment operator.
```

File: src/mlpack/methods/ann/layer/multiply_merge.hpp [hunk 5]
```diff
@@ -50,6 +50,18 @@ class MultiplyMerge
    */
   MultiplyMerge(const bool model = false, const bool run = true);
 
+  //! Copy Constructor
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/multiply_merge.hpp:53]:
```suggestion
  //! Copy Constructor.
```

File: src/mlpack/methods/ann/layer/multiply_merge.hpp [hunk 6]
```diff
@@ -50,6 +50,18 @@ class MultiplyMerge
...
+  MultiplyMerge(const MultiplyMerge& layer);
+
+  //! Move Constructor
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/multiply_merge.hpp:56]:
```suggestion
  //! Move Constructor.
```

File: src/mlpack/methods/ann/layer/multiply_merge.hpp [hunk 7]
```diff
@@ -50,6 +50,18 @@ class MultiplyMerge
...
+  MultiplyMerge(MultiplyMerge&& layer);
+
+  //! Copy assignment operator
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/multiply_merge.hpp:59]:
```suggestion
  //! Copy assignment operator.
```

File: src/mlpack/methods/ann/layer/multiply_merge.hpp [hunk 8]
```diff
@@ -50,6 +50,18 @@ class MultiplyMerge
...
+  MultiplyMerge& operator=(const MultiplyMerge& layer);
+
+  //! Move assignment operator
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/multiply_merge.hpp:62]:
```suggestion
  //! Move assignment operator.
```

**Review by zoq — COMMENTED** [review]
Looks good just some minor style issues.

**Comment by Abilityguy** [comment]
Thanks for the review. 
Made the style fixes.

**Review by zoq — APPROVED** [review]
No more comments from my side, thanks for putting this together.

**Comment by Abilityguy** [comment]
I believe me can merge this.

**Comment by zoq** [comment]
Thanks again for the contribution.

**Pull request merged**